Tribler/Core/Modules/restapi/downloads_endpoint.py

```diff
@@ -102,3 +119,135 @@ def render_GET(self, request):
                              "max_download_speed": download.get_max_speed(DOWNLOAD)}
             downloads_json.append(download_json)
         return json.dumps({"downloads": downloads_json})
+
+
+class DownloadSpecificEndpoint(DownloadBaseEndpoint):
+    """
+    This class is responsible for dispatching requests to perform operations in a specific discovered channel.
+    """
+
+    def __init__(self, session, infohash):
+        DownloadBaseEndpoint.__init__(self, session)
+
+        child_handler_dict = {"remove": DownloadRemoveEndpoint, "stop": DownloadStopEndpoint,
+                              "resume": DownloadResumeEndpoint}
+        for path, child_cls in child_handler_dict.iteritems():
+            self.putChild(path, child_cls(session, bytes(infohash.decode('hex'))))
+
+
+class DownloadRemoveEndpoint(DownloadBaseEndpoint):
+    """
+    This class is responsible for removing downloads from Tribler.
+    """
+
+    def __init__(self, session, infohash):
+        DownloadBaseEndpoint.__init__(self, session)
+        self.infohash = infohash
+
+    def render_DELETE(self, request):
+        """
+        .. http:delete:: /download/(string: infohash)/remove
+
+        A DELETE request to this endpoint removes a specific download from Tribler. You can specify whether you only
+        want to remove the download or the download and the downloaded data using the remove_data parameter.
+
+            **Example request**:
+
+                .. sourcecode:: none
+
+                    curl -X DELETE http://localhost:8085/download/4344503b7e797ebf31582327a5baae35b11bda01/remove
+                    --data "remove_data=1"
+
+            **Example response**:
+
+                .. sourcecode:: javascript
+
+                    {"removed": True}
+        """
+        parameters = http.parse_qs(request.content.read(), 1)
+
+        if 'remove_data' not in parameters or len(parameters['remove_data']) == 0:
+            request.setResponseCode(http.BAD_REQUEST)
+            return json.dumps({"error": "remove_data parameter missing"})
+
+        download = self.session.get_download(self.infohash)
+        if not download:
+            return DownloadRemoveEndpoint.return_404(request)
+
+        remove_data = parameters['remove_data'][0] is True
+        self.session.remove_download(download, removecontent=remove_data)
+
+        return json.dumps({"removed": True})
+
+
+class DownloadStopEndpoint(DownloadBaseEndpoint):
+    """
+    This class is responsible for stopping a download in Tribler.
+    """
+
+    def __init__(self, session, infohash):
+        DownloadBaseEndpoint.__init__(self, session)
+        self.infohash = infohash
+
+    def render_POST(self, request):
+        """
+        .. http:post:: /download/(string: infohash)/stop
```

Comment:
What about `PATCH` with `state=stopped` or similar to the infohash?
http://www.restapitutorial.com/lessons/httpmethods.html